### PR TITLE
Matcher delegate_method supports 'private: true'

### PR DESCRIPTION
**Pull request by pr0d1r2**
Allow to specify `delegate_method` using `with_private` - which is described in documentation:

```ruby
class Account
  delegate :plan, to: :subscription, private: true
end

# RSpec
describe Account do
  it { should delegate_method(:plan).to(:subscription).with_private }
end

# Minitest
class PageTest < Minitest::Test
  should delegate_method(:plan).to(:subscription).with_private
end
```

### CODE REVIEW

**Review by matsales28 — APPROVED** [review]
LGTM. Thanks for the contribution!

File: lib/shoulda/matchers/independent/delegate_method_matcher.rb [hunk 1]
```diff
@@ -202,14 +222,19 @@ def matches?(subject)
...
         end
 
         def description
           string =
             "delegate #{formatted_delegating_method_name} to the " +
             "#{formatted_delegate_object_reader_method_name} object"
 
+          if expects_private_delegation?
+            string << " privately"
+          end
```
Comment (matsales28) [inline lib/shoulda/matchers/independent/delegate_method_matcher.rb:236]: Do you mind changing to single quotes? Rubocop raised at this line in the CI

```suggestion
          if expects_private_delegation?
            string << ' privately'
          end
```

Comment (pr0d1r2) [reply lib/shoulda/matchers/independent/delegate_method_matcher.rb]: Sure thing. Force-pushed updated implementation commit.

**Comment by pr0d1r2** [comment]
@matsales28 Sweet! It is a pure pleasure to work with code specified that well with RSpec 🙏

**Comment by pr0d1r2** [comment]
Found one edge-case that still needs coverage:

```ruby
it { is_expected.to delegate_method(:videos).to(:current_user).with_prefix('user').with_private }
```

Generates error:

```
     NoMethodError:
       private method `user_playlist_videos' called for
```

**Comment by matsales28** [comment]
> Found one edge-case that still needs coverage:
> 
> ```ruby
> it { is_expected.to delegate_method(:videos).to(:current_user).with_prefix('user').with_private }
> ```
> 
> Generates error:
> 
> ```
>      NoMethodError:
>        private method `user_videos' called for
> ```

Yeah, it should also be good to test it in conjunction with the `as` qualifier.

```ruby
  it { is_expected.to delegate_method(:videos).to(:current_user).with_prefix('user').as(:new_videos) }
```

Please let me know if you intend to add the coverage, we can wait for it to merge this PR, otherwise, I can work on improving the coverage, we'll likely need to change the code itself a bit as well to match the error message better.

**Comment by pr0d1r2** [comment]
@matsales28 force-pushed two commits again. The first one specify all combinations. Waiting to see those error messages changes.

**Pull request merged**